modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -532,7 +534,9 @@ def build_index(cls, dataset, partition_ids, index_columns):
 
         # For the second check, let us consider the case where we have an empty dataframe,
         # that has a valid index.
-        if range_index or (len(partition_ids) == 0 and len(column_names_to_read) != 0):
+        if (range_index and filters is None) or (
```

Comment:
While I agree this would be more correct, this is not how it works in Pandas, and I think we should match that behavior. Passing `[]` to pyarrow [results in an error](https://github.com/apache/arrow/blob/19d0daf40c2db5164a6bfefc91019253d8db7157/python/pyarrow/parquet/core.py#L99-L100), though it works in fastparquet. You'll notice in my tests that I included both `None` and `[]` as test cases for this reason.